core/src/main/scala/org/apache/spark/deploy/worker/ExecutorRunner.scala

```diff
@@ -171,6 +172,12 @@ private[deploy] class ExecutorRunner(
       // parent process for the executor command
       builder.environment.put("SPARK_LAUNCH_WITH_SCALA", "0")
 
+      if (shuffleServicePort.isDefined && LocalSparkCluster.get.isDefined) {
```

Comment:
Do we need to add a check for `Utils.isTesting`?

Reply:
No. I think users can also use `LocalSparkCluster` when `Utils.isTesting` is `false`.

Reply:
Yes, good catch!